src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/HaivisionXDecoderCommunicator.java

```diff
@@ -692,31 +870,1955 @@ public boolean validatePortRange(Integer portNumber) {
+				removeUnusedStatsAndControlByEncrypted(stats, advancedControllableProperties, createStream, streamControllingGroup);
+
+				createStream.setSrtSettings(srtSettingEnum.getName());
+				populateCreateStreamControl(stats, advancedControllableProperties, createStream, streamControllingGroup);
+				break;
+			case PASSPHRASE:
+				createStream.setPassphrase(value);
+				addAdvanceControlProperties(advancedControllableProperties,
+						createText(stats, streamControllingGroup + StreamControllingMetric.PASSPHRASE.getName(), value));
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+				break;
+			case REJECT_UNENCRYPTED_CALLERS:
+				RejectUnencrypted rejectUnencryptedCallers = RejectUnencrypted.getByCode(Integer.parseInt(value));
+				createStream.setRejectUnencrypted(rejectUnencryptedCallers.getName());
+
+				addAdvanceControlProperties(advancedControllableProperties, createSwitch(
+						stats, streamControllingGroup + StreamControllingMetric.REJECT_UNENCRYPTED_CALLERS.getName(), rejectUnencryptedCallers.getCode(), DecoderConstant.DISABLE, DecoderConstant.ENABLE));
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+				break;
+			case RTSP_URL:
+				createStream.setAddress(value);
+				populateCreateStreamControl(stats, advancedControllableProperties, createStream, streamControllingGroup);
+				break;
+			case CREATE:
+				performCreateStreamControl();
+				createStream = defaultStream();
+				populateCreateStreamControl(stats, advancedControllableProperties, defaultStream(), streamControllingGroup);
+				isEmergencyDelivery = false;
+				break;
+			case CANCEL:
+				createStream = defaultStream();
+				populateCreateStreamControl(stats, advancedControllableProperties, defaultStream(), streamControllingGroup);
+				isEmergencyDelivery = false;
+				break;
+			default:
+				if (logger.isWarnEnabled()) {
+					logger.warn(String.format("Operation %s with value %s is not supported.", controllableProperty, value));
+				}
+				throw new IllegalStateException(String.format("Operation %s with value %s is not supported.", controllableProperty, value));
+		}
+	}
+
+	/**
+	 * This method is used to perform create stream control
+	 *
+	 * @throws ResourceNotReachableException when fail to send CLI command
+	 */
+	private void performCreateStreamControl() {
+		try {
+			String request = createStream.contributeCommand(CommandOperation.OPERATION_STREAM.getName(), CommandOperation.OPERATION_CREATE.getName());
+			String response = send(request);
+			if (StringUtils.isNullOrEmpty(response) || !response.contains(DecoderConstant.SUCCESSFUL_RESPONSE)) {
+				throw new ResourceNotReachableException(DecoderConstant.SPACE + Deserializer.getErrorMessage(response));
+			}
+		} catch (Exception e) {
+			throw new ResourceNotReachableException(DecoderConstant.DECODER_CONTROL_ERR + DecoderConstant.SPACE + e.getMessage(), e);
```

Comment:
e.getMessage() would be a part of an original error, passed along with the exception, so there's no need in that :) 
Simply passing original one as 2nd parameter should be enough

Reply:
I fixed it